YACS-RCOS/yacs.n
# More Info in Schedule Events

**Issue**
Related to #449

**Database Changes/Migrations**
Added `session_type` and `instructor` to `course_session`.

**Test Modifications**
No.  Columns in `course_session` table do not have any unit test.

**Test Procedure**
- Add some courses, and there will be simplified information in the scheduler (course code, session type and location).
- Click on the schedule event should show the detailed information (course name, instructor, time period, CRN, etc.).
- Click on the course link should redirect to the correct course with correct selected semester.

**Photos**
![PC](https://user-images.githubusercontent.com/11363469/137596756-e8f9b761-3a1f-4dcf-aeb5-0f75a53e2119.png)

**Additional Info**
- Feel free to discuss the approaches of displaying overflowed info in Schedule Events.
- ~~The popover cannot be self-dismissed when clicking to other schedule events, maybe modal should be used here?~~ Fixed
- ~~How to let users know that the schedule events are actually clickable?~~ Moved to `hover`
- Now `hsl()` is implemented in `ColorService`, so colors could reduce saturation and luminosity in dark mode.
- The height of scheduler changed to `80vh` to fill the gap at the bottom.
- Added a new cookie `selectedSemester` to hold the state between refreshes.
